Entity representation (#1393)
spatialos/gdk-for-unity: CHANGELOG.md

- All generated C# enums will now start from 0, being shifted to schema values on serialization and shifted back to C# values on deserialization. [#1412](https://github.com/spatialos/gdk-for-unity/pull/1412)
    - A warning will be generated when enums defined in schema do not start from 0.

> - GameObjectCreation feature module now links entity types with prefabs through EntityRepresentationMapping assets. [#1393](https://github.com/spatialos/gdk-for-unity/pull/1393)

Reviewer: Also feel like there's some additions to the API that should be included here e.g. metaclass interface / component database

- All generated C# enums will now start from 0, being shifted to schema values on serialization and shifted back to C# values on deserialization. [#1412](https://github.com/spatialos/gdk-for-unity/pull/1412)
    - A warning will be generated when enums defined in schema do not start from 0.
- GameObjectCreation feature module now links entity types with prefabs through EntityRepresentationMapping assets. [#1393](https://github.com/spatialos/gdk-for-unity/pull/1393)
    - See the [upgrade guide](UPGRADE_GUIDE.md) for detailed upgrade instructions for this breaking change.

>     - `IEntityGameObjectCreator.PopulateEntityTypeExpectations` Now only needs to specify what SpatialOS components it needs for spawning an entity, such as `Position`.

Reviewer: ```suggestion
    - `IEntityGameObjectCreator.PopulateEntityTypeExpectations` now only needs to specify what SpatialOS components it needs for spawning an entity, such as `Position`.
```

>     - `GameObjectCreationHelperEnableStandardGameObjectCreation` Now requires a non-null `EntityRepresentationMapping` to be passed in.

Reviewer: ```suggestion
    - `GameObjectCreationHelper.EnableStandardGameObjectCreation` now requires a non-null `EntityRepresentationMapping` to be passed in.
```

- Added `Scripting Backend` option dropdown to the Build Configuration UI. [#1411](https://github.com/spatialos/gdk-for-unity/pull/1411)
- Added `Allow Debug` toggle to the Build Configuration UI. [#1413](https://github.com/spatialos/gdk-for-unity/pull/1413)
    - Enabling the option allows .NET debuggers (Rider/Visual Studio) to use breakpoints while in the Unity Player.

> - Added `Authority` type field to the `ComponentDatabase` `IComponentMetaclass`. [#1393](https://github.com/spatialos/gdk-for-unity/pull/1393)

Reviewer: ```suggestion
- Added `Authority` type field to `ComponentDatabase` and `IComponentMetaclass`. [#1393](https://github.com/spatialos/gdk-for-unity/pull/1393)
```

Author: There is no AND here, its a pair
